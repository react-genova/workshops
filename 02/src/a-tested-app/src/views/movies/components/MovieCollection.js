// @flow
import * as React from 'react';
import {List} from 'semantic-ui-react';
import Movie from './Movie';
import type { MovieType } from './types';

type Props = {
    movies: Array<MovieType>,
    updateRate: (id:string, newRate:number) => void
};

const MovieCollection = ({movies,updateRate}: Props) => (
    <List>
        {
            movies != null && movies.map(({id, title, original, year, rating}:MovieType) =>
                <List.Item key={id}>
                    <Movie id={id} title={title} original={original} year={year} rating={rating} updateRate={updateRate} />
                </List.Item>
            )
        }
    </List>
);

export default MovieCollection;