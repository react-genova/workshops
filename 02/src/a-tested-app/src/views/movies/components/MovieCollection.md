### MovieCollection ###

```js
require('semantic-ui-css/semantic.min.css');
const onRate=(newRate)=>{alert(newRate);console.log(newRate)};
const movies = [
    {
        id: "2",
        title: "Ave, Cesare!",
        original: "Hail, Caesar!",
        rating: 3,
        year: 2016
    },
    {
        id: "3",
        title: "Frantic",
        original: "Frantic",
        rating: 3,
        year: 1988
    },
    {
        id: "4",
        title: "Il ritorno dello Jedi",
        original: "Return of the Jedi",
        rating: 4,
        year: 1983
    }
];

<MovieCollection movies={movies} onRate={onRate}/>
```