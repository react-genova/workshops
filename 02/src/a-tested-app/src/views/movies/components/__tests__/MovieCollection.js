//@flow
import React from 'react';
import MovieCollection from '../MovieCollection';
import { shallow } from 'enzyme';

describe('MovieCollection component', () => {

    it('MovieCollection snapshot', () => {
        const updateRate = jest.fn();
        const movies = [
            {
                id: "2",
                title: "Ave, Cesare!",
                original: "Hail, Caesar!",
                rating: 3,
                year: 2016
            },
            {
                id: "3",
                title: "Frantic",
                original: "Frantic",
                rating: 3,
                year: 1988
            },
            {
                id: "4",
                title: "Il ritorno dello Jedi",
                original: "Return of the Jedi",
                rating: 4,
                year: 1983
            }
        ];
        const component = shallow(
            <MovieCollection movies={movies} updateRate={updateRate}/>
        );
        expect(component).toMatchSnapshot();
    });

});