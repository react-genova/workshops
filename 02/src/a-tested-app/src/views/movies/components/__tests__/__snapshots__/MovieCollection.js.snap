// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCollection component MovieCollection snapshot 1`] = `
<List>
  <ListItem
    key="2"
  >
    <Movie
      id="2"
      original="Hail, Caesar!"
      rating={3}
      title="Ave, Cesare!"
      updateRate={[Function]}
      year={2016}
    />
  </ListItem>
  <ListItem
    key="3"
  >
    <Movie
      id="3"
      original="Frantic"
      rating={3}
      title="Frantic"
      updateRate={[Function]}
      year={1988}
    />
  </ListItem>
  <ListItem
    key="4"
  >
    <Movie
      id="4"
      original="Return of the Jedi"
      rating={4}
      title="Il ritorno dello Jedi"
      updateRate={[Function]}
      year={1983}
    />
  </ListItem>
</List>
`;
